### ISSUE TITLE
libreoffice crashes when trying to open or save file

### ISSUE THREAD

**Issue by Profpatsch** [NixOS/nixpkgs#13452]
Update: not yet fully on master, will retry once I am.
Update2: It’s still happening on master.

nixpkgs/master

When hitting `open` or `save as` or similar from inside the program, it crashes. Not when using `save` with an already saved file, so I assume it’s because of the dialogue that should open.

trace:
```
Fatal exception: Signal 5
Stack:
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libuno_sal.so.3(+0x3a890)[0x7f56af6e0890]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libuno_sal.so.3(+0x3ac23)[0x7f56af6e0c23]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libuno_sal.so.3(+0x3ad58)[0x7f56af6e0d58]
/nix/store/n2wxp513rr00f6hr2dy0waqahns49dch-glibc-2.21/lib/libc.so.6(+0x33340)[0x7f56aec31340]
/nix/store/550xwzx0via3yj1mgm2nkvdxqi66vqrr-glib-2.46.2/lib/libglib-2.0.so.0(g_logv+0x1c0)[0x7f56a1ba9100]
/nix/store/550xwzx0via3yj1mgm2nkvdxqi66vqrr-glib-2.46.2/lib/libglib-2.0.so.0(g_log+0x82)[0x7f56a1ba9332]
/nix/store/550xwzx0via3yj1mgm2nkvdxqi66vqrr-glib-2.46.2/lib/libgio-2.0.so.0(+0xbcf82)[0x7f56a219df82]
/nix/store/550xwzx0via3yj1mgm2nkvdxqi66vqrr-glib-2.46.2/lib/libgobject-2.0.so.0(+0x154a9)[0x7f56a1ea54a9]
/nix/store/550xwzx0via3yj1mgm2nkvdxqi66vqrr-glib-2.46.2/lib/libgobject-2.0.so.0(g_object_new_valist+0x3d5)[0x7f56a1ea6ed5]
/nix/store/550xwzx0via3yj1mgm2nkvdxqi66vqrr-glib-2.46.2/lib/libgobject-2.0.so.0(g_object_new+0xd4)[0x7f56a1ea71e4]
/nix/store/06sa7312390qnnl1m6vadml3k8d93icv-gtk+3-3.18.5/lib/libgtk-3.so.0(+0x1a5a1c)[0x7f5690ed5a1c]
/nix/store/06sa7312390qnnl1m6vadml3k8d93icv-gtk+3-3.18.5/lib/libgtk-3.so.0(+0x1a8460)[0x7f5690ed8460]
/nix/store/06sa7312390qnnl1m6vadml3k8d93icv-gtk+3-3.18.5/lib/libgtk-3.so.0(+0x1a35b4)[0x7f5690ed35b4]
/nix/store/550xwzx0via3yj1mgm2nkvdxqi66vqrr-glib-2.46.2/lib/libgobject-2.0.so.0(g_closure_invoke+0x145)[0x7f56a1e9ffc5]
/nix/store/550xwzx0via3yj1mgm2nkvdxqi66vqrr-glib-2.46.2/lib/libgobject-2.0.so.0(+0x21bc1)[0x7f56a1eb1bc1]
/nix/store/550xwzx0via3yj1mgm2nkvdxqi66vqrr-glib-2.46.2/lib/libgobject-2.0.so.0(g_signal_emit_valist+0xf86)[0x7f56a1eba476]
/nix/store/550xwzx0via3yj1mgm2nkvdxqi66vqrr-glib-2.46.2/lib/libgobject-2.0.so.0(g_signal_emit_by_name+0x4ea)[0x7f56a1ebab9a]
/nix/store/550xwzx0via3yj1mgm2nkvdxqi66vqrr-glib-2.46.2/lib/libgobject-2.0.so.0(g_object_set_property+0x20b)[0x7f56a1ea7f3b]
/nix/store/550xwzx0via3yj1mgm2nkvdxqi66vqrr-glib-2.46.2/lib/libgobject-2.0.so.0(g_object_set_valist+0x21b)[0x7f56a1ea741b]
/nix/store/550xwzx0via3yj1mgm2nkvdxqi66vqrr-glib-2.46.2/lib/libgobject-2.0.so.0(g_object_set+0xcf)[0x7f56a1ea7c3f]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libvclplug_gtk3lo.so(+0x48e80)[0x7f5691690e80]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libvclplug_gtk3lo.so(+0x4a112)[0x7f5691692112]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libvcllo.so(_ZN11Application16createFilePickerERKN3com3sun4star3uno9ReferenceINS3_17XComponentContextEEE+0x48)[0x7f56a9360e48]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libsvtlo.so(+0x30525d)[0x7f56aa8c325d]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libuno_cppuhelpergcc3.so.3(+0x47540)[0x7f56ad056540]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libuno_cppuhelpergcc3.so.3(+0x4f4fb)[0x7f56ad05e4fb]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libuno_cppuhelpergcc3.so.3(+0x8dce4)[0x7f56ad09cce4]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libuno_cppuhelpergcc3.so.3(+0x93aa0)[0x7f56ad0a2aa0]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libuno_cppuhelpergcc3.so.3(+0x8bd54)[0x7f56ad09ad54]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libsfxlo.so(+0x26b870)[0x7f56abc64870]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libsfxlo.so(_ZN4sfx216FileDialogHelperC2EslRKN3rtl8OUStringEs14SfxFilterFlagsS5_S4_RKN3com3sun4star3uno8SequenceIS2_EE+0x9d)[0x7f56abc670ad]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libsfxlo.so(+0x349487)[0x7f56abd42487]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libsfxlo.so(+0x34dd50)[0x7f56abd46d50]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libsfxlo.so(+0x3711ed)[0x7f56abd6a1ed]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libsfxlo.so(+0x1dc859)[0x7f56abbd5859]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libsfxlo.so(+0x1d968c)[0x7f56abbd268c]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libsfxlo.so(+0x21dc1c)[0x7f56abc16c1c]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libsfxlo.so(+0x2204a9)[0x7f56abc194a9]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/../program/libfwklo.so(+0x21eecf)[0x7f5686e13ecf]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libvcllo.so(_ZN4Menu6SelectEv+0x77)[0x7f56a90ee1d7]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libvcllo.so(+0x229253)[0x7f56a915a253]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libvcllo.so(_ZN17SalGenericDisplay21DispatchInternalEventEv+0x71)[0x7f56a93d4111]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libvclplug_gtk3lo.so(+0x3c629)[0x7f5691684629]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libvclplug_gtk3lo.so(+0x3c6a1)[0x7f56916846a1]
/nix/store/550xwzx0via3yj1mgm2nkvdxqi66vqrr-glib-2.46.2/lib/libglib-2.0.so.0(g_main_context_dispatch+0x13d)[0x7f56a1ba2a3d]
/nix/store/550xwzx0via3yj1mgm2nkvdxqi66vqrr-glib-2.46.2/lib/libglib-2.0.so.0(+0x48dd8)[0x7f56a1ba2dd8]
/nix/store/550xwzx0via3yj1mgm2nkvdxqi66vqrr-glib-2.46.2/lib/libglib-2.0.so.0(g_main_context_iteration+0x2c)[0x7f56a1ba2e7c]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libvclplug_gtk3lo.so(+0x3bb67)[0x7f5691683b67]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libvcllo.so(_ZN11Application5YieldEv+0x51)[0x7f56a935e7a1]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libvcllo.so(_ZN11Application7ExecuteEv+0x45)[0x7f56a9360a75]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libsofficeapp.so(+0x289fc)[0x7f56af4369fc]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libvcllo.so(+0x4330c1)[0x7f56a93640c1]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libvcllo.so(_Z6SVMainv+0x22)[0x7f56a9364112]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/libsofficeapp.so(soffice_main+0xa2)[0x7f56af45ab92]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/soffice.bin[0x4008db]
/nix/store/n2wxp513rr00f6hr2dy0waqahns49dch-glibc-2.21/lib/libc.so.6(__libc_start_main+0xf5)[0x7f56aec1e995]
/nix/store/lxk07d81vp7ak2dqb9ailwwxm4bxamb1-libreoffice-5.1.0.3/lib/libreoffice/program/soffice.bin[0x400919]
```

cc @viric

**Comment by vcunat** [comment]
That's exactly how missing gtk3 gsettings schemas manifest, although a more informative error was always printed, so it might be something else.

**Comment by Profpatsch** [comment]
The derivation switched from `gtk` to `gtk3` in the last version; that should probably be the reason for this @7c6f434c.

**Comment by 7c6f434c** [comment]
>The derivation switched from `gtk` to `gtk3` in the last version; that should probably be the reason for this @7c6f434c.

Yeah, didn't remember about the schema part. I will probably try to make
a wrapper around LibreOffice for such things…

**Comment by Profpatsch** [comment]
@7c6f434c Bear in mind that it’s completely unusable atm *and* in master. I advise to rewind the change as fast as possible. I was already bitten by that.

**Comment by 7c6f434c** [comment]
Yes, I have probably underestimated severity because I used LibreOffice just fine after the update (and I am currently trying to build a fix without splitting LibreOffice into many-staged build). 

OK, pushed the temporary revert, hopefully it works

**Comment by skrzyp** [comment]
Still not fixed, happened to me right now, worst moment today.

I think we should care about stability of office suites and tools. It's very important because people often wants to read/modify some files when they're in hurry, and suddenly, **BOOM** - Libreoffice says "you will not edit this today, go away and die painfully" because package maintainer haven't checked it before commiting.

**Comment by 7c6f434c** [comment]
>Still not fixed, happened to me right now, worst moment today.

Cannot reproduce on master.

What is the LibreOffice store path you have?

>I think we should care about stability of office suites and tools. It's very important because people often wants to read/modify some files when they're in hurry, and suddenly, **BOOM** - Libreoffice says "you will not edit this today, go away and die painfully" because package maintainer haven't checked it before commiting.

The part about editing is false, this is the part that I did check even 
initially and that doesn't require dialogs to work… Reading definitely
doesn't need dialogs.

**Comment by skrzyp** [comment]
> What is the LibreOffice store path you have?


````
[skrzyp@ono-sendai:~]$ ls /nix/store/ | grep libreoffice
b2xgca7yf5nq8j2jw6nl1hklvkwxjs3w-libreoffice-help-5.1.0.3.tar.xz.lock
gagg2wcmqb9ba5qr4qqsaax8ir5bpy11-libreoffice-5.1.0.3.tar.xz.drv
lc47l09i2rbpp2m7n2h80jbpc43yr5w9-libreoffice-help-5.1.0.3.tar.xz.drv
lqnlij473yizqlz13yl2rl5iljf81nik-libreoffice-translations-5.1.0.3.tar.xz.drv
pkxvhf0frawh0y8kygkc827xrcxxprgy-glm-0.9.4.6-libreoffice.zip.lock
r0d6gzvxn07nkf3n3vdy72q2z4aw2gf1-glm-0.9.4.6-libreoffice.zip.drv
vnj9iydg2pgzgl575ky21r89cf2szrks-libreoffice-translations-5.1.0.3.tar.xz.lock
x5hyzyp8a3jpfdjh7agh2klg4svdiqkg-libreoffice-5.1.0.3.drv
ylmshk43jsdfi6dwfybj9a5yjhq5px2y-libreoffice-5.1.0.3.tar.xz.lock
zhvj8rhmknys2hhxgwl18y1bygm7qjgd-libreoffice-5.1.0.3
````

**Comment by 7c6f434c** [comment]
You have only one LibreOffice output path, which means it is the same
version you originally reported the issue with (and Hydra build list
confirms that).

Fresh build of master LibreOffice (available on https://cache.nixos.org/
and the one I am using) is 
/nix/store/wn6q6axpr6frk9mwwvwnxg5m9zizcwgs-libreoffice-5.1.0.3 which 
does have a working open-file dialog.

**Comment by skrzyp** [comment]
But I'm using unstable channel, which is upgraded daily by `nix-upgrade` service. So, what else I should do to acquire the same hash of LO like you're using?

**Comment by joachifm** [comment]
@skrzyp the unstable channel is only upgraded when the Hydra job completes, it's not uncommon for it to lag behind by several weeks.

**Comment by skrzyp** [comment]
But he said that certain LO expression result is available on cache.nixos.org, so I think hydra has processed it before.

**Comment by joachifm** [comment]
@skrzyp sure, but nixos-unstable won't update before the entire jobset completes.

**Comment by joachifm** [comment]
@skrzyp you could always just download/install the closure you want directly from hydra

**Comment by skrzyp** [comment]
How? It's not documented I think.

**Comment by joachifm** [comment]
@skrzyp e.g., https://hydra.nixos.org/build/32657170

**Comment by 7c6f434c** [comment]
>But I'm using unstable channel, which is upgraded daily by `nix-upgrade` service. So, what else I should do to acquire the same hash of LO like you're using?

So we again have channel not updating for more than 2 days (the earliest
non-aborted post-fix build was 2 days ago)? Pity.

```
nix-store -r /nix/store/wn6q6axpr6frk9mwwvwnxg5m9zizcwgs-libreoffice-5.1.0.3 --indirect --add-root /path/to/lo-symlink
/path/to/lo-symlink/bin/libreoffice
```
should be the way that works regardless of how you install the rest of the packages.

**Comment by Profpatsch** [comment]
@7c6f434c Maybe you can add VM tests for these kinds of errors? Chromium uses these kinds of tests afaik.

**Comment by 7c6f434c** [comment]
>@7c6f434c Maybe you can add VM tests for these kinds of errors? Chromium uses these kinds of tests afaik.

I have no idea how to do VM tests, and I am out of date re: NixOS tests 
as I don't use its boot sequence anymore (the FS layout I use is the 
same, though).

Also, my use patterns apparently do not intersect with the others' use
patterns…

**Comment by Profpatsch** [comment]
You don’t open file dialogues from inside LibreOffice? :)

**Comment by 7c6f434c** [comment]
>You don’t open file dialogues from inside LibreOffice? :)

Actually, I never use them. When I want to open a file, I usually want 
to open this specific file (from shell or from Firefox or whatever), and
LibreOffice is just an implementation detail… So LibreOffice gets the 
file as a command-line parameter…

**Comment by viric** [comment]
@7c6f434c oh, I'm interested in your boot sequence. Is it public?

**Comment by 7c6f434c** [comment]
>@7c6f434c oh, I'm interested in your boot sequence. Is it public?

Yes, https://github.com/7c6f434c/7c6f434c-configurations/blob/master/initrd-to-read-boot/pseudo-system.nix

(how to go completely off topic in an issue discussion in three 
comments)

**Comment by skrzyp** [comment]
@7c6f434c what.

**Comment by 7c6f434c** [comment]
@skrzyp As I said, my comment was completely offtopic — I was just answering @viric 's question.

**Comment by skrzyp** [comment]
@7c6f434c I know, but it was about that linked .nix file. Seriously, what the hell is happening there? :o

**Comment by 7c6f434c** [comment]
@skrzyp It just builds what I use as `/run/current-system/` Puts daemon launcher scripts in their places, does setuid wrapping etc.

**Comment by skrzyp** [comment]
I'll love some detailed paper about this, of course if you want to spare time for that.

**Comment by 7c6f434c** [comment]
>I'll love some detailed paper about this, of course if you want to spare time for that.

I'd love to have some time to clean it up and actually make
configurable…

And currently there is precisely nothing to write a paper about: this is
all trivial if you have read any of the GNU/Linux boot step-by-step 
walkthroughs (and the Nix manual).

Splitting it and making it configurable not in the NixOS module style,
but in the NixPkgs style may be interesting, but requires some effort.

**Comment by grahamc** [comment]
This appears to be resolved in 16.03. Can anyone else confirm?

**Comment by vcunat** [comment]
I confirm this, both on 16.03 and recent master.

**Comment by bburdette** [comment]
I filed a bug re this on the libreoffice tracker.  Was advised to use this command, which works for me on unstable (which still has the bug for others and not just me I hope):

`SAL_USE_VCLPLUGIN=gtk soffice`

**Comment by teto** [comment]
I still have the bug.

**Comment by 7c6f434c** [comment]
@teto I think you need to provide much more details… (and probably you don't have the same bug but a similar one)

Are you on NixOS? What Nixpkgs revision? What is the LibreOffice store path? Is it enough to start LibreOffice, create a new Writer document and press Ctrl-s immediately after it is shown?

**Comment by teto** [comment]
Libreoffice crashed when I tried to open a file but I was too fast the bug happened again when trying to open a popup with firefox. I'll open an issue when I get more time/info. It is a bug that I have met on and off a few times for the past months though not in the last weeks hence my surprise.

**Comment by 7c6f434c** [comment]
Environment variables (XDG stuff, GTK stuff, DBus stuff) could also be relevant…

**Comment by Profpatsch** [comment]
> I filed a bug re this on the libreoffice tracker. Was advised to use this command, which works for me on unstable (which still has the bug for others and not just me I hope):

Do you think it might make sense to wrap libreoffice so it’s always executed with this environment variable? Or could this break the settings of some users?

**Comment by bburdette** [comment]
@Profpatsch sorry I'm a bit out of the loop now, having been on ubuntu for the past year or so.  I looked on the libreoffice tracker and failed to find the bug I filed, so no help there either.  A little bash wrapper would probably work for me since I'm on xmonad and I initiate everything from the commandline or dmenu.  But the situation may be more complicated for folks that use gnome menus where the libreoffice apps are broken out into the different apps.
